colosseum/declaration.py

```diff
@@ -51,10 +51,15 @@ def validated_property(name, choices, initial):
     try:
         initial = choices.validate(initial)
     except ValueError:
+        # The initial value might be a OtherProperty or Custom class with a value method
         try:
-            # The initial value might be a OtherProperty or Custom class with a value method
-            initial.value
-        except AttributeError:
+            # Check it has a value attribute
+            value_attr = getattr(initial, 'value')
+
+            # Check the value attribute is a callable
+            assert callable(value_attr)
```

Comment:
This isn't sufficient - assert statements are turned into no-ops if you turn on bytecode optimization. You need to actually check and raise a ValueError. It's also a different error case between "doesn't have a value attribute", and "has a value attribute, but it isn't callable.

Reply:
Fixed!